### PR TITLE
Feature/migration

### PR DESCRIPTION
**Pull request by shahnCM**
1. Created "master" branch
2. Generated Laravel Project in "master" & pushed
3. Created "develop" branch from "master" and pushed
4. Created "feature/migration"  from "develop" 
5. Created all the migrations and pushed it to "feature/migration"
6. Now asking for PR

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -17,10 +17,14 @@ public function up()
             $table->id();
             $table->string('name');
             $table->string('email')->unique();
+            $table->string('phone')->unique();
```
Comment (me-shaon) [inline database/migrations/2014_10_12_000000_create_users_table.php]: No need to make it 'unique'.

File: database/migrations/2020_11_04_144429_create_publications_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('publications', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('image');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144429_create_publications_table.php]: nullable.

File: database/migrations/2020_11_04_144429_create_publications_table.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            $table->string('description');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144429_create_publications_table.php]: nullable.

File: database/migrations/2020_11_04_144958_create_books_table.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('books', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('slug');
+            $table->string('image');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144958_create_books_table.php]: nullable

File: database/migrations/2020_11_04_144958_create_books_table.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('books', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('slug');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144958_create_books_table.php]: unique.

File: database/migrations/2020_11_04_144958_create_books_table.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            $table->string('image');
+
+            // $table->bigInteger('publication_id');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144958_create_books_table.php]: remove commented code.

File: database/migrations/2020_11_04_144958_create_books_table.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            $table->foreignId('publication_id')->constrained('publications');
+            
+            $table->bigInteger('isbn_number');
+            $table->bigInteger('total_copies');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144958_create_books_table.php]: default 0.

File: database/migrations/2020_11_04_144958_create_books_table.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            $table->bigInteger('available_copies');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144958_create_books_table.php]: default 0.

File: database/migrations/2020_11_04_144958_create_books_table.php [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('books');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_144958_create_books_table.php:42]: Drop foreign key first, then drop the table.

Comment (me-shaon) [reply database/migrations/2020_11_04_144958_create_books_table.php:42]: Fix it in other places too.

File: database/migrations/2020_11_04_145901_create_authors_table.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('authors', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('image');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_145901_create_authors_table.php]: nullable.

File: database/migrations/2020_11_04_145901_create_authors_table.php [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            $table->string('description');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_145901_create_authors_table.php]: nullable.

File: database/migrations/2020_11_04_150008_create_author_book_table.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('author_book', function (Blueprint $table) {
+            $table->id();
+
+            // $table->bigInteger('author_id');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_150008_create_author_book_table.php]: remote commented code.

Comment (me-shaon) [reply database/migrations/2020_11_04_150008_create_author_book_table.php]: Fix in other places too.

File: database/migrations/2020_11_04_150440_create_genres_table.php [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('genres', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('slug');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_150440_create_genres_table.php]: unique.

File: database/migrations/2020_11_04_160917_create_loan_requests_table.php [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        Schema::create('loan_requests', function (Blueprint $table) {
+            $table->id();
+
+            // $table->bigInteger('user_id');
+            $table->foreignId('user_id')->constrained('users');
+            
+            // $table->bigInteger('status_changed_by');
+            $table->foreignId('status_changed_by')->constrained('users');
+            
+            $table->string('status');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_160917_create_loan_requests_table.php]: set default status.

File: database/migrations/2020_11_04_160917_create_loan_requests_table.php [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+            $table->string('reason');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_160917_create_loan_requests_table.php]: nullable.

File: database/migrations/2020_11_04_160925_create_return_requests_table.php [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+        Schema::create('return_requests', function (Blueprint $table) {
+            $table->id();
+
+            // $table->bigInteger('user_id');
+            $table->foreignId('user_id')->constrained('users');
+            
+            // $table->bigInteger('status_changed_by');
+            $table->foreignId('status_changed_by')->constrained('users');
+            
+            $table->string('status');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_160925_create_return_requests_table.php]: set default status.

File: database/migrations/2020_11_04_160925_create_return_requests_table.php [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+            $table->string('reason');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_160925_create_return_requests_table.php]: nullable.

File: database/migrations/2020_11_04_174749_create_book_copies_table.php [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('book_copies', function (Blueprint $table) {
+            $table->id();
+            $table->string('uuid');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_174749_create_book_copies_table.php]: unique.

File: database/migrations/2020_11_04_174749_create_book_copies_table.php [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+
+            // $table->bigInteger('book_id');
+            $table->foreignId('book_id')->constrained('books');
+            
+            $table->string('edition');
+            $table->string('condition');
+            $table->string('description');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_174749_create_book_copies_table.php]: nullable.

File: database/migrations/2020_11_04_174749_create_book_copies_table.php [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+            $table->boolean('is_available');
```
Comment (me-shaon) [inline database/migrations/2020_11_04_174749_create_book_copies_table.php]: set default value.

**Review by me-shaon — CHANGES_REQUESTED** [review]

**Comment by shahnCM** [comment]
Resolved Commented Points

**Pull request closed without merging**